test/Microsoft.AspNet.Mvc.Core.Test/LocalRedirectResultTest.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc.Abstractions;
+using Microsoft.AspNet.Mvc.Infrastructure;
+using Microsoft.AspNet.Mvc.Routing;
+using Microsoft.AspNet.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class LocalRedirectResultTest
+    {
+        [Fact]
+        public void LocalRedirectResult_Constructor_WithParameterUrl_SetsResultUrlAndNotPermanent()
+        {
+            // Arrange
+            var url = "/test/url";
+
+            // Act
+            var result = new LocalRedirectResult(url);
+
+            // Assert
+            Assert.False(result.Permanent);
+            Assert.Same(url, result.Url);
+        }
+
+        [Fact]
+        public void LocalRedirectResult_Constructor_WithParameterUrlAndPermanent_SetsResultUrlAndPermanent()
+        {
+            // Arrange
+            var url = "/test/url";
+
+            // Act
+            var result = new LocalRedirectResult(url, permanent: true);
+
+            // Assert
+            Assert.True(result.Permanent);
+            Assert.Same(url, result.Url);
+        }
+
+        [Theory]
+        [InlineData(null, "~/Home/About", "/Home/About")]
+        [InlineData("/", "~/Home/About", "/Home/About")]
+        [InlineData("/", "~/", "/")]
+        [InlineData("", "~/Home/About", "/Home/About")]
+        [InlineData("/myapproot", "~/", "/myapproot/")]
+        [InlineData("", "/Home/About", "/Home/About")]
+        [InlineData("/myapproot", "/test", "/test")]
+        public void Execute_ReturnsExpectedValues(
```

Comment:
A good reason to have one case and nothing more. Overtesting just makes everything harder to maintain